File: src/main/java/navrat/name/vivicta/model/Transaction.java
package navrat.name.vivicta.model;

import static jakarta.persistence.EnumType.STRING;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "chs_transactions")
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String accountName;
    private BigDecimal amount;
    private String counterpartyAccountName;
    private Integer creditDebitFlag;
    private Integer currencyCode;
    private String descr1;
    private String messageForRecipient;
    private LocalDate transactionDate;
    private Integer transactionNumber;
    private LocalDate transactionSentDate;
    private Long variableSymbol;
    @Enumerated(STRING)
    private ProcessingStatus processingStatus;
    @Version
    private int version;
}

File: ui-client/src/types/transaction.ts
export interface Transaction {
  id: string;
  transactionSentDate: string;
  transactionNumber: number;
  amount: number;
  currencyCode: number | string;
  variableSymbol: number | null;
  counterpartyAccountName?: string;
  messageForRecipient?: string;
  processingStatus: 'PENDING' | 'PENDING_MANUAL' | 'AUTO_PROCESSED' | 'IGNORE' | 'MANUALY_FIXED';
  isDirty?: boolean;
}
